### PR TITLE
Generalize `-rotate` for `|n|` greater than `(length list)`.

### LINKED ISSUE #284: (-rotate 7 '(1 2 3)) should work by moding 7 by length of list

### PR DESCRIPTION
**Pull request by leungbk**
Closes #284.

This works but is probably not idiomatic; I'd appreciate suggestions on how to tidy it up.

### CODE REVIEW

File: dash.el [hunk 1]
```diff
@@ -921,9 +921,11 @@ See also: `-drop'"
   "Rotate LIST N places to the right.  With N negative, rotate to the left.
 The time complexity is O(n)."
   (declare (pure t) (side-effect-free t))
-  (if (> n 0)
-      (append (last list n) (butlast list n))
-    (append (-drop (- n) list) (-take (- n) list))))
+  (and list
```
Comment (Fuco1) [inline dash.el]: It's better to use `when` if there is only one condition followed by a body.

File: dash.el [hunk 2]
```diff
@@ -921,9 +921,11 @@ See also: `-drop'"
...
+       (let* ((list-len (length list))
```
Comment (Fuco1) [inline dash.el]: This could probably be called just `len`, it's understandable from context

File: dash.el [hunk 3]
```diff
@@ -921,9 +921,11 @@ See also: `-drop'"
...
+              (n-mod-len (mod n list-len))
```
Comment (Fuco1) [inline dash.el]: It's nice here that it gets rid of the special case for negatives as it automatically produces the corresponding positive offset <3

**Comment by Fuco1** [comment]
@leungbk Not bad at all, just minor style issues and I'm happy to merge!

**Comment by leungbk** [comment]
@Fuco1 Thanks for your suggestions, I've updated the PR accordingly.

**Comment by Fuco1** [comment]
@leungbk 	Thanks! :rocket:

**Pull request merged**